numba/np/arraymath.py

```diff
@@ -4116,11 +4116,12 @@ def impl(ap1, ap2, mode, direction):
         # For "direction", +1 to write the return values out in order 0->N
         # -1 to write them out N->0.
 
-        if not (mode == Mode.VALID or mode == Mode.FULL):
-            raise ValueError("Invalid mode")
-
         n1 = len(ap1)
         n2 = len(ap2)
+
+        if n1 < n2:
+            raise ValueError("'len(ap1)' must greate than 'len(ap2)'")
```

Comment:
```suggestion
            raise ValueError("'len(ap1)' must greater than 'len(ap2)'")
```

Also a good idea to add a test for this error, so that we have coverage.